<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  
  
  <link rel="shortcut icon" href="../img/favicon.ico">
  <title>Feature - CropGBM</title>
  <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700" />

  <link rel="stylesheet" href="../css/theme.css" />
  <link rel="stylesheet" href="../css/theme_extra.css" />
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css" />
  
  <script>
    // Current page data
    var mkdocs_page_name = "Feature";
    var mkdocs_page_input_path = "Feature.md";
    var mkdocs_page_url = null;
  </script>
  
  <script src="../js/jquery-2.1.1.min.js" defer></script>
  <script src="../js/modernizr-2.8.3.min.js" defer></script>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
  <script>hljs.initHighlightingOnLoad();</script> 
  
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
    <div class="wy-side-scroll">
      <div class="wy-side-nav-search">
        <a href=".." class="icon icon-home"> CropGBM</a>
        <div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="..">Home</a>
                    </li>
                </ul>
                <ul class="current">
                    <li class="toctree-l1 current"><a class="reference internal current" href="./">Feature</a>
    <ul class="current">
    <li class="toctree-l2"><a class="reference internal" href="#use-lightgbm-algorithm-to-predict-phenotypic-values-and-select-snps">Use LightGBM algorithm to predict phenotypic values and select SNPs</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#introducing-t-sne-method-to-reduce-dimensionality-of-genotype-data">Introducing t-SNE method to reduce dimensionality of genotype data</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#introducing-optics-method-to-cluster-data">Introducing OPTICS method to cluster data</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#plot-the-distribution-histogram-of-parameters-of-the-genotype-data">Plot the distribution histogram of parameters of the genotype data</a>
    </li>
    </ul>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../QuickStart/">QuickStart</a>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../Tutorial/">Tutorial</a>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../Parameters/">Parameters</a>
                    </li>
                </ul>
      </div>
    </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="..">CropGBM</a>
      </nav>

      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="..">Docs</a> &raquo;</li>
    
      
    
    <li>Feature</li>
    <li class="wy-breadcrumbs-aside">
      
    </li>
  </ul>
  
  <hr/>
</div>
          <div role="main">
            <div class="section">
              
                <h1 id="features">Features</h1>
<p>This section details the characteristics of the methods used by CropGBM in data processing, and briefly introduces the principles of these methods.<br/><br/></p>
<h2 id="use-lightgbm-algorithm-to-predict-phenotypic-values-and-select-snps">Use LightGBM algorithm to predict phenotypic values and select SNPs</h2>
<p>At present, methods such as GBLUP are commonly used to predict phenotype values. Calculate the correlation matrix between samples from the entire genome marker or genotype, and then use the mixed linear model (MLM) to calculate the estimated breeding value (EBV) of the samples. But GBLUP also has many shortcomings:</p>
<ol>
<li>Difficult to capture complex non-linear relationships between genotypes. When faced with complex population structure prediction problems, its cross-population prediction ability is not ideal.</li>
<li>The kinship matrix is not extensible. If the training population changes, the kinship matrix needs to be recalculated. At the same time, the computation time of the kinship matrix increases dramatically with the increase of the sample size of the training population.</li>
<li>Unable to solve the multi-classification problem. Many genotype-to-phenotype prediction tasks require classification prediction.<br/><br/></li>
</ol>
<p>With the development of genotype detection, phenotype detection and other technologies, the cost of sample acquisition has continued to decline. The genotype and phenotype data have accumulated rapidly. The sample size has grown from hundreds to thousands. As GBLUP and other methods need to calculate the correlation matrix between samples, its computing performance decreases significantly with the increase of the population size. CropGBM uses the LightGBM algorithm, which is a large number of optimized Gradient Boosting Decision Tree (GBDT) algorithms, as the kernel of phenotype prediction. When processing large batches of sample data, compared to GBLUP, the lightgbm algorithm can be used without losing accuracy:</p>
<ul>
<li>Dramatically reduce memory consumption.</li>
<li>Significantly improve prediction speed and support GPU-accelerated model training.</li>
<li>Support discontinuous phenotype value prediction<br/><br/></li>
</ul>
<p>CropGBM can also mine phenotype-related SNP sites according to the weight of SNP in the model. Generally, the larger the weight of SNP, the stronger the correlation with phenotype. It is recommended that users compare the weights of each SNP provided by CropGBM with the GWAS results to eliminate false positive results in their respective methods and improve their reliability.<br/><br/></p>
<h2 id="introducing-t-sne-method-to-reduce-dimensionality-of-genotype-data">Introducing t-SNE method to reduce dimensionality of genotype data</h2>
<p>The t-distributed stochastic neighbor embedding (t-SNE) algorithm is a nonlinear dimensionality reduction algorithm for mining high-dimensional data. The data points are mapped from high-dimensional to low-dimensional in the form of conditional probability, and t-distribution is used to cleverly solve the congestion caused by dimensionality reduction of high-dimensional data, so that the dimensionality reduction results retain the local and global structure of the data. Linear dimensionality reduction algorithms such as PCA cannot ensure that the data structure of non-linear high-dimensional data can be mapped correctly after dimensionality reduction. On the other hand, when the data is reduced to two dimensions, the t-SNE algorithm re-arranges the data on a two-dimensional plane, while the PCA algorithm selects the two dimensions from the high-dimensional space that can best show the difference in data to display the data. Therefore, the t-SNE algorithm is better at displaying dimensionality reduction data.</p>
<p>Because t-SNE algorithm has quadratic time and space complexity, it requires higher system resources. Before using the t-SNE algorithm to reduce the dimensionality of the program, the program first reduced the dimensionality of the SNP data through a sliding window. The SNP data is divided into multiple groups according to the sliding window size set by the user. The sliding window size is the number of SNP. The sum of the SNP values in each group is used as the value of the feature in the new dimension to achieve the purpose of reducing system resource consumption and increasing SNP feature value richness.<br/><br/></p>
<h2 id="introducing-optics-method-to-cluster-data">Introducing OPTICS method to cluster data</h2>
<p>The ordering points to identify the clustering structure (OPTICS) algorithm is a density-based clustering algorithm. Finding high-density regions separated by low-density regions in the data set and dividing the separated high-density regions into clusters, it can find any shape of the group structure. However, the K-Means algorithm and hierarchical clustering algorithm commonly used in the current research are distance-based clustering algorithms. They perform well on spherical data sets. But when the data set is non-spherical structure, the clustering effect of the above algorithm is not good. On the other hand, K-Means algorithm requires users to provide the number of clusters in advance. But sometimes researchers may not know the population structure of the sample set. The OPTICS algorithm can automatically determine the number of clusters based on the dense distribution of data in the sample set, helping users to discover the population structure and sort out the relationships between samples.<br/><br/></p>
<h2 id="plot-the-distribution-histogram-of-parameters-of-the-genotype-data">Plot the distribution histogram of parameters of the genotype data</h2>
<p>The pre-processing module of the program supports statistics on the sample heterozygosity rate, genotype heterozygosity rate, minor allele frequency, genotype deletion rate, and sample deletion rate of the input genotype data and outputs it as a histogram. It is convenient for users to understand the overall situation of the data and formulate subsequent research methods.</p>
              
            </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="../QuickStart/" class="btn btn-neutral float-right" title="QuickStart">Next <span class="icon icon-circle-arrow-right"></span></a>
      
      
        <a href=".." class="btn btn-neutral" title="Home"><span class="icon icon-circle-arrow-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
    
  </div>

  Built with <a href="https://www.mkdocs.org/">MkDocs</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
      
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" aria-label="versions">
    <span class="rst-current-version" data-toggle="rst-current-version">
      
      
        <span><a href=".." style="color: #fcfcfc;">&laquo; Previous</a></span>
      
      
        <span style="margin-left: 15px"><a href="../QuickStart/" style="color: #fcfcfc">Next &raquo;</a></span>
      
    </span>
</div>
    <script>var base_url = '..';</script>
    <script src="../js/theme.js" defer></script>
      <script src="../search/main.js" defer></script>
    <script defer>
        window.onload = function () {
            SphinxRtdTheme.Navigation.enable(true);
        };
    </script>

</body>
</html>
