<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  
  <link rel="shortcut icon" href="../img/favicon.ico">
  <title>Parameters - CropGBM</title>
  <link href='https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700' rel='stylesheet' type='text/css'>

  <link rel="stylesheet" href="../css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../css/theme_extra.css" type="text/css" />
  <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css">
  
  <script>
    // Current page data
    var mkdocs_page_name = "Parameters";
    var mkdocs_page_input_path = "Parameters.md";
    var mkdocs_page_url = null;
  </script>
  
  <script src="../js/jquery-2.1.1.min.js" defer></script>
  <script src="../js/modernizr-2.8.3.min.js" defer></script>
  <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
  <script>hljs.initHighlightingOnLoad();</script> 
  
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
      <div class="wy-side-nav-search">
        <a href=".." class="icon icon-home"> CropGBM</a>
        <div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
	<ul class="current">
	  
          
            <li class="toctree-l1">
		
    <a class="" href="..">Home</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../Feature/">Feature</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../QuickStart/">QuickStart</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../Tutorial/">Tutorial</a>
	    </li>
          
            <li class="toctree-l1 current">
		
    <a class="current" href="./">Parameters</a>
    <ul class="subnav">
            
    <li class="toctree-l2"><a href="#parameters">Parameters</a></li>
    
        <ul>
        
            <li><a class="toctree-l3" href="#default-module">DEFAULT module</a></li>
        
            <li><a class="toctree-l3" href="#genotype-data-preprocessing-module">Genotype data preprocessing module</a></li>
        
            <li><a class="toctree-l3" href="#phenotyp-data-preprocessing-module">Phenotyp data preprocessing module</a></li>
        
            <li><a class="toctree-l3" href="#population-structure-analysis-module">Population structure analysis module</a></li>
        
            <li><a class="toctree-l3" href="#training-model-snp-selection-and-phenotype-prediction-module">Training model, SNP selection and phenotype prediction module</a></li>
        
            <li><a class="toctree-l3" href="#lightgbm-parameters">LightGBM parameters</a></li>
        
        </ul>
    

    </ul>
	    </li>
          
        </ul>
      </div>
      &nbsp;
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="..">CropGBM</a>
      </nav>

      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="..">Docs</a> &raquo;</li>
    
      
    
    <li>Parameters</li>
    <li class="wy-breadcrumbs-aside">
      
    </li>
  </ul>
  <hr/>
</div>
          <div role="main">
            <div class="section">
              
                <h1 id="parameters">Parameters</h1>
<p>In this section, the parameters of each module are divided into four types: Boolean, necessary, semi-necessary, and optional.</p>
<ul>
<li>Boolean parameters are divided into assigned and unassigned. When assigning, the program will execute the function corresponding to the parameter; otherwise, the program will not execute the corresponding function.</li>
<li>If the necessary parameter is not assigned a value, an error will be reported when the module it belongs to is called.</li>
<li>If the semi-necessary parameter is not assigned a value, an error will be reported when calling a specific function under the module.</li>
<li>If optional parameters are not assigned, certain functions cannot be used, but no error will be reported.</li>
</ul>
<p>Except for DEFAULT module parameters, other parameters are only used when the owning module is called. <br/> <br/></p>
<h2 id="default-module">DEFAULT module</h2>
<p>The following parameters will be used in each module: <br/>
<code>-c</code>,<code>--config-path</code>: The path of the configuration file. The file indicates the value of each parameter that the program needs to call. When the parameters in the file overlap with the command line, the program will use the command line value as the standard. <br/>
<code>-o</code>,<code>--output-folder</code>: The output path of the program result. Default is the current working path. <br/> <br/></p>
<h2 id="genotype-data-preprocessing-module">Genotype data preprocessing module</h2>
<ul>
<li><strong>Boolean</strong></li>
</ul>
<p><code>-pg</code>, <code>--preprocessed-geno</code>: Call genotype data preprocessing module, optional values [all, filter]. The 'all' indicates that the program will call the complete pre-processing module, including: statistical and visual data basics, screening out non-compliant sample individuals and SNP sites based on the deletion rate, MAF, and filling the missing data of SNP based on high-frequency genotypes, removing redundant SNP according to r ^ 2, extracting specific genotype data function of specific sample according to SNPID or sampleID. The 'filter' indicates that the program will call part of the preprocessing module, which only includes the specific extraction of specific samples according to SNPID or sampleID<br/>
<code>--recode</code>: Recode the genotype data file, transform the encoding method from base encoding to additive encoding.</p>
<ul>
<li><strong>necessary</strong></li>
</ul>
<p><code>--fileprefix</code>: Genotype data filename prefix. <br/>
<code>--fileformat</code>: Genotype data file format, optional values [ped, bed, vcf, bcf].</p>
<ul>
<li><strong>optional</strong></li>
</ul>
<p><code>--plink-path<code>ï¼šSpecify plink path. Default is plink<br/>
<code>--remove-sampleid-path</code>: The file path of the sampleID to be deleted. The format is two columns with spaces or tabs as separators. The first column is familyID and the second column is within-familyID. The program will delete the corresponding sample data in the genotype file according to sampleID. <br/>
<code>--keep-sampleid-path</code>: The file path of the sampleID to be extracted. The file format is the same as <code>--remove-sampleid-path</code>. <br/>
<code>--extract-snpid-path</code>: The file path of the SNPID to be extracted, the format is a list of SNPID. The program will extract the corresponding SNP data from the genotype file based on the SNPID. <br/>
<code>--exclude-snpid-path</code>: The file path of the SNPID to be deleted. The file format is the same as <code>--extract-snpid-path</code>. <br/>
<code>--snpmaxmiss</code>: The maximum miss rate of SNP. Screen out SNP with missing rate exceeding threshold. Default is 0.05 <br/>
<code>--samplemaxmiss</code>: Maximum sample miss rate. Screen out samples whose missing rate exceeds the threshold. Default is 0.05 <br/>
<code>--maf-max</code>: The minimum allele frequency. SNP with minimum allele frequency less than the threshold was screened out. Default is 0.01 <br/>
<code>--r2-cutoff</code>: Equivalent to the<code>--indep-pairwise</code> parameter in PLINK. Default is 0.8 <br/> <br/></p>
<h2 id="phenotyp-data-preprocessing-module">Phenotyp data preprocessing module</h2>
<ul>
<li><strong>Boolean</strong></li>
</ul>
<p><code>-pp</code>,<code>--preprocessed-phe</code>: Call the phenotype data preprocessing module. <br/>
<code>--phe-norm</code>: Call the phenotype data normalization function. <br/>
<code>--phe-plot</code>: Call phenotype data visualization function. <br/>
<code>--phefile-header</code>: The phenotype data file contains the header. <br/>
<code>--ppgroupfile-header</code>: The groupID data file contains the header.</p>
<ul>
<li><strong>necessary</strong></li>
</ul>
<p><code>--phefile-path</code>: The phenotype data file path.</p>
<ul>
<li><strong>semi-necessary</strong></li>
</ul>
<p><code>--ppexsampleid-path</code>: The file path of the sampleID to be extracted. The file content is only a list of sampleID. <br/>
<code>--ppgroupfile-path</code>: GroupID data file path. <br/>
<code>--ppgroupfile-sep</code>: GroupID data file separator. Default is ',' <br/>
<code>--ppgroupid-column</code>: The column index that stores the groupID. This parameter is only used when the number of ppgroupfile columns is greater than 2. Default is 2 <br/>
<code>--num2wordfile-path</code>: The file path of transformation table of integer and phenotype. This parameter is only used when <code>--phe-recode=num2word</code>. <br/></p>
<ul>
<li><strong>optional</strong></li>
</ul>
<p><code>--phefile-sep</code>: phenotype data file separator. Default is ',' <br/>
<code>--phe-column</code>: The column index in the phenotype data file that stores phenotype. This parameter is only called when the number of columns in the phenotype file is greater than 2. Default is 2 <br/>
<code>--phe-recode</code>: Call the phenotype data recoding function, optional values [word2num, num2word]. The 'word2num' means re-encoding the phenotype data into continuous non-negative integer form. The 'num2word' means re-encoding the continuous non-negative integer form into phenotype data (need to provide a conversion table corresponding to integer and phenotype). When performing classification tasks, Lightgbm only accepts consecutive integers with example labels [0, N). If the training sample comes from 5 groups, [0, 1, 2, 3, 4] is required as the label of the 5 groups, but this usually does not match the actual groupID. Using this parameter, the program can implement a reversible conversion between sample labels and [0, N) consecutive integers. Provide compatible phenotype data for downstream classification tasks. <br/>
<code>--norm-mode</code>: Normalization algorithm, optional values [0-1, max-min, z-score]. Default is z-score <br/> <br/></p>
<h2 id="population-structure-analysis-module">Population structure analysis module</h2>
<ul>
<li><strong>Boolean</strong></li>
</ul>
<p><code>-s</code>,<code>--structure</code>: Call the population structure analysis module. <br/>
<code>--structure-plot</code>: Call the population structure plot function. A total of 2 or 3 scatter plots are displayed to show the results of dimensionality reduction and clustering. <br/>
<code>--sgroupfile-header</code>: The groupID data file contains the header.</p>
<ul>
<li><strong>necessary</strong></li>
</ul>
<p><code>--genofile-path</code>: The path of the genotype data file. The pre-processed results are often used as input.</p>
<ul>
<li><strong>semi-necessary</strong></li>
</ul>
<p><code>--n-clusters</code>: the number of clusters. This parameter is only used when <code>--cluster-mode = kmeans</code>. <br/>
<code>--sgroupfile-path</code>: GroupID data file path. <br/>
<code>--sgroupfile-sep</code>: GroupID data file separator. Default is ',' <br/>
<code>--sgroupfile-column</code>: The column index that stores groupID. This parameter is only used when the number of <code>--sgroupfile-path</code> file columns is greater than 2.</p>
<ul>
<li><strong>optional</strong></li>
</ul>
<p><code>--redim-mode</code>: Dimension reduction algorithm, optional value [tsne, pca]. Default is pca <br/>
<code>--pca-explained-var</code>: The parameter value is an integer greater than 1 or a decimal between 0-1. When the parameter value is an integer, it indicates the dimension of the data after pca dimensionality reduction; when the parameter value is a decimal, it indicates the amount of variation that can be explained by the data after pca dimensionality reduction. Default is 0.95. This parameter is only used when <code>--redim-mode = pca</code> <br/>
<code>--window-size</code>: Sliding window size in dimensionality reduction. Because the calculation speed of the t-SNE dimensionality reduction algorithm decreases significantly with increasing dimensions, the program performs a preliminary dimensionality reduction on the data by sliding windows. Default is 20. This parameter is only used when <code>--redim-mode = tsne</code>. <br/>
<code>--tsne-dim</code>: The dimension of the t-SNE algorithm after dimensionality reduction. Default is 2. This parameter is only used when <code>--redim-mode = tsne</code>. <br/>
<code>--cluster-mode</code>: Dimension reduction algorithm, optional value [tsne, pca]. Default is Kmeans.<br/>
<code>--optics-min-samples</code>: The parameter value is an integer greater than 1 or a decimal between 0-1. When the parameter value is an integer, it indicates the minimum number of samples required to form the core point; when the parameter value is a decimal number, it indicates the ratio of the minimum number of samples required to form the core point to the total sample number. This parameter is only used when <code>--cluster-mode = optics</code>. Default is 0.025 <br/>
<code>--optics-xi</code>: The minimum value of the reachable distance gradient required for the aggregation class. This parameter is only used when <code>--cluster-mode = optics</code>. Default is 0.05 <br/>
<code>--optics-min-cluster-size</code>: The parameter value is an integer greater than 1 or a decimal between 0-1. When the parameter value is an integer, it indicates the minimum number of samples required for the aggregation. When the parameter value is a decimal number, it indicates the minimum number of samples required for the aggregation. This parameter is only used when <code>--cluster-mode = optics</code>. Default is 0.03 <br/> <br/></p>
<h2 id="training-model-snp-selection-and-phenotype-prediction-module">Training model, SNP selection and phenotype prediction module</h2>
<ul>
<li><strong>Boolean</strong></li>
</ul>
<p><code>-e</code>,<code>--engine</code>: Call the training model, SNP selection and phenotype prediction module. <br/>
<code>-t</code>,<code>--train</code>: Call the training model function. <br/>
<code>-cv</code>: Call the cross-validation function. <br/>
<code>-p</code>,<code>--predict</code>: Call the phenotype prediction function. <br/>
<code>-sf</code>,<code>--select-feature</code>: Call SNP selection function. <br/>
<code>--bygain-boxplot</code>: Draws a boxplot that keeps changing as the SNP is added to the model. This parameter is used when <code>-sf</code> is specified.</p>
<ul>
<li><strong>semi-necessary</strong></li>
</ul>
<p><code>--traingeno</code>: The path to the genotype data file for the training set. The file separator is ',', the first line is SNPID, and the first column is sampleID. This parameter is used when <code>-t</code> or<code>-cv</code> is specified. <br/>
<code>--trainphe</code>: The path to the phenotype data file for the training set. The file separator is ',', including the header, and the first column is sampleID information. This parameter is used when <code>-t</code> or<code>-cv</code> is specified. <br/>
<code>--testgeno</code>: The path to the genotype data file for the test set. File format is the same as <code>--traingeno</code>. This parameter is used when <code>-p</code> is specified. <br/>
<code>--modelfile-path</code>: The path of the LightGBM model file. The program uses this model to complete the phenotype prediction of the test set. This parameter is used when <code>-p</code> is specified.</p>
<ul>
<li><strong>optional</strong></li>
</ul>
<p><code>--validgeno</code>: The path to the genotype data file for the validation set. File format is the same as <code>--traingeno</code>. This parameter is used when <code>-t</code> is specified. <br/>
<code>--validphe</code>: Path to the sampleID file of the validation set sample. File format is the same as <code>--traingeno</code>. This parameter is used when <code>-t</code> is specified. <br/>
<code>--init-model-path</code>: The path of the initial model file. The LightGBM algorithm will continue to train on this model. Commonly used for batch training of big data. This parameter is used when <code>-t</code> or<code>-cv</code> is specified. <br/>
<code>--min-detal</code>: The minimum percentage of model accuracy improvement per training. The program returns the number of times the model is trained when the accuracy is less than this threshold and the accuracy of the cross-validation of the model. When <code>objective=regression</code>, the accuracy is the mean of the mse between the predicted result and the true value. This parameter is used when <code>-cv</code> is specified. Default is 0.5 <br/>
<code>--cv-times</code>: Number of repetitions for cross-validation. This parameter is used when <code>-sf</code> is specified. Default is 5 <br/>
<code>--cv-nfold</code>: The number of folds per cross-validation. This parameter is used when <code>-cv</code> is specified. Default is 5 <br/>
<code>--gain-min</code>: calculate the total gain of each SNP in the model, and use the product of the maximum gain and<code>--gain-min</code> as the threshold. When the total gain of the SNP in the model is less than the threshold, this SNP will not be selected. This parameter is used when <code>-sf</code> and<code>--bygain-boxplot</code> are specified. Default is 0.05 <br/>
<code>--colorbar-max</code>: The product of the maximum gain of each SNP in each tree and<code>--colorbar-max</code> will be used as the maximum value of colorbar in the heat map. This parameter is used when <code>-sf</code> is specified. Default is 0.6 <br/></p>
<h2 id="lightgbm-parameters">LightGBM parameters</h2>
<p><code>--learning-rate</code>: The learning rate of the model. Default is 0.1 <br/>
<code>--num-leaves</code>: The maximum number of leaves per decision tree. Default is 10 <br/>
<code>--num-threads</code>: The number of CPU threads available for core module operation calls. Default is 0, which is the default number of threads in OpenMP. <br/>
<code>--min-data-in-leaf</code>: The minimum number of samples in each leaf. Default is 1 <br/>
<code>--objective</code>: The type of purpose for model training. Default is 'regression' <br/>
<code>--device-type</code>: The device type called by training model, optional value ['cpu', 'gpu']. Default is 'cpu' <br/>
<code>--max-depth</code>: The maximum depth of the decision tree. If the value is &lt;= 0, there is no maximum depth limit. Default is -1 <br/>
<code>--feature-fraction</code>: The proportion of features used in training the model to the total number of features. Default is 1 <br/>
<code>--verbosity</code>: Controls the complexity of information output during model training. &lt;0: Fatal, = 0: Error (Warning), = 1: Info,&gt; 1: Debug. Default is 0 <br/>
<code>--num-class</code>: The number of classes in the training set. This parameter is only called for multi-class tasks, that is, <code>--objective: multiclass</code>. Default is 1 <br/>
<code>--num-boost-round</code>: The number of training (iterations) of the model. Default is 100 <br/>
<code>--early-stopping-rounds</code>: This parameter should not be called only if<code>-cv</code> or <code>-t</code> has a validation set. After every N rounds of training (N is the value of the user-defined parameter), the program will calculate the prediction accuracy of the model on the validation set. If the prediction accuracy no longer improves, the model will stop training immediately. This parameter can be used to prevent model training from overfitting. Default is 20 <br/>
<code>--verbose-eval</code>: This parameter should not be called only when<code>-cv</code> or <code>-t</code> has a validation set. After each N rounds of training (N is the value of the user-defined parameter), the program will calculate the prediction accuracy of the model on the validation set and output it. Default is 10</p>
              
            </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
      
        <a href="../Tutorial/" class="btn btn-neutral" title="Tutorial"><span class="icon icon-circle-arrow-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
    
  </div>

  Built with <a href="http://www.mkdocs.org">MkDocs</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
      
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" style="cursor: pointer">
    <span class="rst-current-version" data-toggle="rst-current-version">
      
      
        <span><a href="../Tutorial/" style="color: #fcfcfc;">&laquo; Previous</a></span>
      
      
    </span>
</div>
    <script>var base_url = '..';</script>
    <script src="../js/theme.js" defer></script>
      <script src="../search/main.js" defer></script>

</body>
</html>
