<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  
  <link rel="shortcut icon" href="../img/favicon.ico">
  <title>QuickStart - CropGBM</title>
  <link href='https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700' rel='stylesheet' type='text/css'>

  <link rel="stylesheet" href="../css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../css/theme_extra.css" type="text/css" />
  <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css">
  
  <script>
    // Current page data
    var mkdocs_page_name = "QuickStart";
    var mkdocs_page_input_path = "QuickStart.md";
    var mkdocs_page_url = null;
  </script>
  
  <script src="../js/jquery-2.1.1.min.js" defer></script>
  <script src="../js/modernizr-2.8.3.min.js" defer></script>
  <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
  <script>hljs.initHighlightingOnLoad();</script> 
  
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
      <div class="wy-side-nav-search">
        <a href=".." class="icon icon-home"> CropGBM</a>
        <div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
	<ul class="current">
	  
          
            <li class="toctree-l1">
		
    <a class="" href="..">Home</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../Feature/">Feature</a>
	    </li>
          
            <li class="toctree-l1 current">
		
    <a class="current" href="./">QuickStart</a>
    <ul class="subnav">
            
    <li class="toctree-l2"><a href="#quickstart">QuickStart</a></li>
    
        <ul>
        
            <li><a class="toctree-l3" href="#installation">Installation</a></li>
        
            <li><a class="toctree-l3" href="#parameter-configuration">Parameter configuration</a></li>
        
            <li><a class="toctree-l3" href="#genotype-data-pre-processing">Genotype data pre-processing</a></li>
        
            <li><a class="toctree-l3" href="#phenotype-data-pre-processing">Phenotype data pre-processing</a></li>
        
            <li><a class="toctree-l3" href="#population-structure-analysis">Population structure analysis</a></li>
        
            <li><a class="toctree-l3" href="#building-models-and-snp-selection">Building models and SNP selection</a></li>
        
            <li><a class="toctree-l3" href="#phenotype-prediction">Phenotype prediction</a></li>
        
        </ul>
    

    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../Tutorial/">Tutorial</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../Parameters/">Parameters</a>
	    </li>
          
        </ul>
      </div>
      &nbsp;
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="..">CropGBM</a>
      </nav>

      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="..">Docs</a> &raquo;</li>
    
      
    
    <li>QuickStart</li>
    <li class="wy-breadcrumbs-aside">
      
    </li>
  </ul>
  <hr/>
</div>
          <div role="main">
            <div class="section">
              
                <h1 id="quickstart">QuickStart</h1>
<p>This section briefly introduces the functions and usage of CropGBM to help users get started quickly. For a detailed introduction to CropGBM, see <a href="../Tutorial/">Tutorial</a><br/></p>
<p>Test data download address 1: <a href="https://gitee.com/cau-xyt/cropgbm_tutorial_data/tree/master">https://gitee.com/cau-xyt/cropgbm_tutorial_data/tree/master</a><br/>Test data download address 2: <a href="https://github.com/YuetongXU/CropGBM_Tutorial-data">https://github.com/YuetongXU/CropGBM_Tutorial-data</a></p>
<p><br/></p>
<h2 id="installation">Installation</h2>
<pre><code>$ tar -zxf CropGBM.tar.gz

# Install Python package dependencies of CropGBM: setuptools, wheel, numpy, scipy, pandas, scikit-learn, lightgbm, matplotlib, seaborn
$ pip install setuptools wheel numpy scipy pandas scikit-learn lightgbm matplotlib seaborn

# Install external dependencies of CropGBM: PLINK 1.90 
$ wget s3.amazonaws.com/plink1-assets/plink_linux_x86_64_20191028.zip
$ mkdir plink_1.90
$ unzip plink_linux_x86_64_20191028.zip -d ./plink_1.90

# Add CropGBM, PLINK to the system environment variables for quick use:
$ vi ~/.bashrc
export PATH="/userpath/CropGBM:$PATH"
export PATH="/userpath/plink1.90:$PATH"
$ source ~/.bashrc
</code></pre>
<p><br/></p>
<h2 id="parameter-configuration">Parameter configuration</h2>
<p>CropGBM supports two parameter assignment forms: 'configuration file' and 'command line', of which 'configuration file' is required. CropGBM will read the values of the parameters in the configuration file first, and then read the values of the parameters in the command line. When a parameter is assigned by both methods at the same time, CropGBM uses the parameter value in the command line as a reference and ignores the parameter value in the configuration file.</p>
<pre><code># CropGBM reads the values of each parameter in the configuration file (-c config_path) and calls the genotype data preprocessing module (-pg all)
$ cropgbm -c config_path -o ./gbm_result/ -pg all

# CropGBM ignores the fileformat value in the configuration file and uses ped as a reference
$ cropgbm -c config_path -o ./gbm_result/ -pg all --fileformat ped
</code></pre>
<p><strong> NOTE: </strong> If the program does not work, try adding python before the program name. E.g. <em>$ python cropgbm -c config_path -pg all</em> <br/><br/></p>
<h2 id="genotype-data-pre-processing">Genotype data pre-processing</h2>
<p>The functions of the genotype data pre-processing module include: extracting the genotype data of specific sample ID and SNPID, displaying the SNP deletion rate, heterozygosity in the form of histogram and genotype recoding. Provide data and acceptable file formats for downstream analysis of the program. CropGBM currently supports ped, bed, vcf, and bcv input formats for genotype files.</p>
<pre><code># Call the genotype data pre-processing module to count and display the missing rate and heterozygosity rate of genotype data
$ cropgbm -c config_path -o ./gbm_result/ -pg all --fileprefix genofile --fileformat ped
</code></pre>
<p><br/></p>
<h2 id="phenotype-data-pre-processing">Phenotype data pre-processing</h2>
<p>The functions of the phenotype data pre-processing module include: extracting phenotype data of specific sample ID, SNPID, groupID, phenotype normalization and phenotype recoding. It also supports displaying the distribution of data in the form of histograms or boxplots.</p>
<pre><code># Call the phenotype data preprocessing module (-pp) for normalization (--phe-norm)
$ cropgbm -c config_path -o ./gbm_result/ -pp --phe-norm --phefile-path phefile.txt --phefile-header --phe-column 4

# Extract phenotypic data based on the groupID (--ppgroupfile-path groupfile.txt) and display it as a boxplot
$ cropgbm -c config_path -o ./gbm_result/ -pp --phe-plot --phefile-path phefile.txt --phefile-header --phe-column 4 --ppgroupfile-path phefile.txt --ppgroupid-column 2 --ppgroupfile-header
</code></pre>
<p><br/></p>
<h2 id="population-structure-analysis">Population structure analysis</h2>
<p>The population structure analysis module can analyze the population structure of a sample based on genotype data. CropGBM supports dimensionality reduction using t-SNE or PCA methods and clustering using OPTICS or Kmeans methods. It also supports displaying the population structure of samples in the form of scatter plots.</p>
<pre><code># Call the population structure analysis module (-s) to cluster and display samples based on genotype data (--structure-plot)
$ cropgbm -c config_path -o ./gbm_result/ -s --structure-plot --genofile-path genofile_filter.geno --redim-mode pca --cluster-mode kmeans --n-clusters 30
</code></pre>
<p><br/></p>
<h2 id="building-models-and-snp-selection">Building models and SNP selection</h2>
<p>The model training module is mainly written based on the lightGBM algorithm. To improve the accuracy of the model, it is recommended to provide a validation set to assist in tuning. If there is no validation set, cross validation can be used to select appropriate parameter values. CropGBM select SNP based on the gain of each SNP in the training model. It also supports the use of boxplot and heatmap to show the importance of the selected SNP.</p>
<pre><code># Cross-validation (-e -cv)
$ cropgbm -c config_path -o ./gbm_result/ -e -cv --traingeno train.geno --trainphe train.phe

# Build model (-e -t). If there is no validation set data, the --validgeno and --validphe parameters can be omitted.
$ cropgbm -c config_path -o ./gbm_result/ -e -t --traingeno train.geno --trainphe train.phe --validgeno valid.geno --validphe valid.phe

# SNP selection (-e -t -sf), showing changes in model prediction accuracy (--bygain-boxplot)
$ cropgbm -c config_path -o ./gbm_result/ -e -t -sf --bygain-boxplot --traingeno train.geno --trainphe train.phe
</code></pre>
<p><br/></p>
<h2 id="phenotype-prediction">Phenotype prediction</h2>
<p>The phenotype prediction module uses the trained model to predict the phenotype of the test set.</p>
<pre><code># Phenotype prediction (-e -p)
$ cropgbm -c config_path -o ./gbm_result/ -e -p --testgeno test.geno --modelfile-path train.lgb_model
</code></pre>
              
            </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="../Tutorial/" class="btn btn-neutral float-right" title="Tutorial">Next <span class="icon icon-circle-arrow-right"></span></a>
      
      
        <a href="../Feature/" class="btn btn-neutral" title="Feature"><span class="icon icon-circle-arrow-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
    
  </div>

  Built with <a href="http://www.mkdocs.org">MkDocs</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
      
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" style="cursor: pointer">
    <span class="rst-current-version" data-toggle="rst-current-version">
      
      
        <span><a href="../Feature/" style="color: #fcfcfc;">&laquo; Previous</a></span>
      
      
        <span style="margin-left: 15px"><a href="../Tutorial/" style="color: #fcfcfc">Next &raquo;</a></span>
      
    </span>
</div>
    <script>var base_url = '..';</script>
    <script src="../js/theme.js" defer></script>
      <script src="../search/main.js" defer></script>

</body>
</html>
